src/main/resources/assets/neoforge/lang/en_us.json

```diff
@@ -194,6 +194,8 @@
   "neoforge.chatType.system": "%1$s",
 
   "pack.neoforge.description": "NeoForge data/resource pack",
+  "pack.neoforge.experimental.name": "NeoForge mod experiments data pack",
+  "pack.neoforge.experimental.description": "Enables experimental mod items",
```

Comment:
I'd suggest "mod features" instead of "mod items". Better be vague here.

Reply:
Perhaps: `Enables experimental mod features that may be incomplete`

Reply:
I feel like neither of those tell a user what will actually be enabled, which... is a bit of a problem. A generic flag for "enable God-know-what experimental features from _every single mod you have installed_" isn't very useful or intuitive to users.

Reply:
This is my big worry and pain point. I still feels experimental features should be a flag or config *per-feature* for easy enabling and disabling along with better understanding of what the user is getting into. 1 broken feature in one mod can easily make a pack unplayable with no way to disable it without disabling the other experimental features you DO want.

Reply:
To clarify, this is not to be turned on by Packs, but on world creation. So this is specifically someone turning it on for their world, if they update a mod and it crashes, that is no different than them updating a mod with no feature flag enabled and it crashing. 

This is intended for "DO NOT turn this on unless you know precisely what you are doing." Granular control is something else entirely that has no current infrastructure. I a feature flag per mod could exist, cool: but that is very outside of scope nor is the FeatureFlag currently built to be able to support that.

Reply:
Ok... but who is actually going to use it then?

If this is to just test features, use a second branch in you repo. That's what branches are for. Build a jar with that test branch and give it to your testers to test with. Now you don't have to fiddle with pack finder events and moving json files around when you go to release the feature. Just merge the feature branch to your main branch and release.

If this is to have random users try random stuff from mods, not dedicated testers, then that should have granular control imo. This smells like a solution in search of a problem that doesn't exist.

Reply:
> If this is to have random users try random stuff from mods
I'll also add that if the contents locked behind this catch all flag introduce a bug, that world is stuck in the broken state until the bug is fixed, as nbt editing is the only way to disable the flag.

Reply:
> > If this is to have random users try random stuff from mods
> 
> I'll also add that if the contents locked behind this catch all flag introduce a bug, that world is stuck in the broken state until the bug is fixed, as nbt editing is the only way to disable the flag.

Or... downgrade the mod that you updated like normal? 🤔

Reply:
And what happens if all version of that mod for the mc version has the broken feature? Or downgrading removes a stable feature that isn't present in older version? This is adding more problems to deal with for the public that this flag is going to be exposed to.

Reply:
> Ok... but who is actually going to use it then?
> 
> If this is to just test features, use a second branch in your repo. That's what branches are for. Build a jar with that test branch and give it to your testers to test with. Now you don't have to fiddle with pack finder events and moving json files around when you go to release the feature. Just merge the feature branch to your main branch and release.
> 
> If this is to have random users try random stuff from mods, not dedicated testers, then that should have granular control imo. This smells like a solution in search of a problem that doesn't exist.

Instead of having to recreate the entire infrastructure for a testing suite, a feature flag that is NOT `bundles` or `Villager trading tweaks` is ideal. Before it made sense to use the 1.21 feature flag, but all of these that exist are dependent on vanilla's life cycle for said feature. None of them are slated to be persistent, which is desired from this.

Even if it was, the current response is to use "bundles" for the most part, which I'd argue is a worse "catch-all" for a user.

Reply:
> This is intended for "DO NOT turn this on unless you know precisely what you are doing."
My point here is that a user _cannot_ know "precisely what they are doing" with this flag, because it enables a bunch of experimental features from countless mods in any given installation. The lack of granularity means that users _can't_ make an informed decision about whether they ought to be turning on the flag or no -- it's not just "test this one mod's experimental feature", it's "test any experimental features of any mods that happen to be installed, all at once". I would say this whole PR makes no sense unless there's a way for modders to list to users what things from their mod the flag will enable, at the point where the user enables the flag.

Reply:
I disagree. While I can certainly see the need for better expression towards the user, "Not intended for gameplay"  should be clear enough for someone to NOT turn on in a world they want to play, and if they want to explore a potentially broken set of features (of all the mods they have installed) they may do so turning that on. Expansion on that is possible, but you are limited in how you can say this given the screen real estate. It definitely needs to be expressed to mod pack makers to not enable.

### The current solutions today:
- Tag the feature as either in `bundles` or `villager_trades`, both much less clear to a user that something incomplete or likely to crash is now being turned on. 
- Do a separate branch on a repository. That adds a burden on both the developer's time (non-trivial) and the player to understand "beta" is not meant to be played. See EnderIO slated as beta and not intended to be used, where most packs and users just put it in blindly thinking it is ready.
- Just don't do feature flags. This is arguably worse and a terrible argument.

Previous to 1.21 there was a `1.21 Experimental options` flag but as indicated above, the life cycle of the flag is tied to arbitrary features of said flag and are not persistent. This indicated the features were not yet ready. Vanilla literally already did what we are asking here, we are just wanting a persistent flag.

### Solutions I am in favor of:
- A persistent catch all Feature Flag for a Test World Creation as this pr is addressing. Yes, any mod installed using this flag would be turned on, but at the time of turning on the flag, the user should expect it to have a high chance of breaking. If they didn't turn on a flag, and added/updated a mod that broke, that is fundamentally no different here than what it is without the flag. This should be indicating that no gameplay should be done with this flag on, and will turn on experimental features of EVERY mod installed. Forgecraft would be an example of when the flag would be turned on.
- A granular control of feature elements. The problem here, is that this one keeps getting tossed around as if it is trivial to build, and no, it is not just "config" as there are a lot of components missing from what feature flags enable/disable. There is no current built in infrastructure for this, so if you are going to toss this one around, you need to build something substantial or show a PR that solves this. Both systems can work hand in hand and not mutually exclusive.  
- Feature flags per mod: ideal scenario, but as far as I understand with what is synced for flags, it is not possible at the moment.

Of these, so far, no solutions exist that are as simple as adding a "Hey Don't Turn this On Unless Testing and not playing". This alleviates the technical burden of reinventing the ENTIRE feature flag system that vanilla has for a list of elements a mod may not finish but wants to make a build. Yes, users will turn this on. Yes users will report bugs on it; however, users already do this, with much less that are not bugs.

Alternatively, If you are up for it, we could add a few more persistent flags that are `alpha` `beta` `experimental` to express more intent to the user and dev.
- `alpha`: Broken, will crash nearly 100%
- `beta`: Not intended to crash, but is not outside of realm of possibility
- `experimental`: Going into prod soon.
The problem with doing this, is that when one moves from one to the other, it may be turned off based on what the player has turned on (unless there is a dependency chain of feature flags I don't know about).

Reply:
The issue isn't about when people _shouldn't_ turn it on, it's about when people _should_ turn it on. As currently written, this mod only gives users the option to turn on every experimental feature from every single currently installed mod. Which is useless for both users _and_ modders -- for users, it means they're getting a _bunch_ of potentially-buggy features all at once instead of the one specific one they're interested in testing out. For modders, it means that any bugs people find with the experimental features -- and we want bug reports for those so we can fix it before fully shipping the feature -- will be polluted by potential bugs from any number of _other_ mods' experimental features. "Test every single experimental feature from every installed mod" is _not_ a use case that users or modders generally want. In fact, many modders would be likely to reject outright a bug report from such a scenario, even if the issue is one of mod compat, unless the user could reproduce it with the other mods' experimental stuff turned _off_.

Per-mod or per-mod-feature feature flags are not, to my understanding, possible with how vanilla handles feature flags. Correct me if I'm wrong, as this would be the optimal solution. With that in mind, it seems like the proper solution is to make it easy for mods to configure experimental features that can be turned on independently of one another, that does not use feature flags. At present, it seems like this should be fully possible through the use of a startup-loaded config. The same "this is experimental and will break stuff" warning as can be put in a feature flag, can also be put in a mod config. If there are "components missing from what feature flags enable/disable" in what configs do -- you have not made them clear. Creative mode tab contents, loaded data, etc. -- all can be controlled from configs!

At the very least -- if folks _do_ decide to go the direction of this feature flag approach, which seems to me like the worst possible approach for this -- then I would say there needs to be a way for users to see -- when enabling the experimental mod features flag -- what mods are using that flag and what they're doing with it. In other words, there has to be a description somewhere there that mods can add info to the contents of somehow.

Reply:
Translations have been updated but the values of these unfortunately can not provide all the information everyone is requesting.